src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1321,6 +1342,100 @@ private bool OnlyWhitespaceOrCommentsAfterExtent(InternalScriptExtent extent)
             return true;
         }
 
+        internal bool IsPipeContinuance(IScriptExtent extent)
+        {
+            return extent.EndOffset < _script.Length && PipeContinuanceAfterExtent(extent);
+        }
+
+        private bool PipeContinuanceAfterExtent(IScriptExtent extent)
+        {
+            // If the first non-newline, non-whitespace, non-comment, non-backtick, non-semi-colon character
+            // following the newline is a pipe, we have a pipe continuance.
+            bool lastNonWhitespaceIsNewline = true;
+            int i = extent.EndOffset;
+
+            // Since some token pattern matching looks for multiple characters (e.g. newline, block comment,
+            // or line continuance), we stop searching at _script.Length - 1 and perform one additional check
+            // after the while loop. This avoids having to compare i + 1 against the script length in multiple
+            // locations inside the loop.
+            while (i < _script.Length - 1)
+            {
+                char c = _script[i];
+
+                if (c.IsWhitespace())
+                {
+                    i++;
+                    continue;
+                }
+
+                if (c == '\n')
+                {
+                    if (lastNonWhitespaceIsNewline)
+                    {
+                        // blank or whitespace-only lines are not allowed in automatic line continuance
+                        return false;
+                    }
+
+                    lastNonWhitespaceIsNewline = true;
+                    i++;
+                    continue;
+                }
+                else if (c == '\r')
+                {
+                    if (lastNonWhitespaceIsNewline)
+                    {
+                        // blank or whitespace-only lines are not allowed in automatic line continuance
+                        return false;
+                    }
+
+                    lastNonWhitespaceIsNewline = true;
+                    i += _script[i + 1] == '\n' ? 2 : 1;
+                    continue;
+                }
+
+                lastNonWhitespaceIsNewline = false;
+
+                if (c == '#')
+                {
+                    // SkipLineComment will return the position after the comment end
+                    // which is either at the end of the file, or a cr or lf.
+                    i = SkipLineComment(i + 1);
+                    continue;
+                }
+
+                if (c == '<' && _script[i + 1] == '#')
+                {
+                    i = SkipBlockComment(i + 2);
+                    continue;
+                }
+
+                if (c == '`')
```

Comment:
Given that already as of this PR, the following two scenarios are very different, I don't think there's anything wrong with this.

```powershell
PS> Get-Verb
| % Verb

PS> Get-Verb
-Verb get
```

Reply:
My concern is on the explicit line continuation token.
- First, I don't see why we want to allow backtick when deciding for the pipe continuance. For one-line comment and command block, absolutely, but I don't see a use case where allowing the backtick is useful. It would be great if you have an example.
- Second, the explicit continuation token make things a bit confusing. If `| % verb` on the next line of a backtick works, then a user would assume `-Verb get` on the next line of a backtick works too, because they are following the continuation token.

Reply:
I agree with @daxian-dbw. Having backtick support here just makes for poorly written code in this case, and since I changed this to only support continuance when the lines are contiguous with comments in the middle, the backtick support here does more harm than good. Plus, the method is much easier to digest with that removed.

I updated the code with the backtick support removed from that method, and added a specific Pester test to make sure that it is not supported when using implicit line continuance.

Reply:
On a related note, the sooner we get this done, the sooner I can add a PR with an experimental feature to support automatic line continuance with parameter names. :) Just waiting for this first since it has significant parser/tokenizer changes.